sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -478,11 +482,13 @@ case class ArrayFilter(
     }
   }
 
-  @transient lazy val (elementVar, indexVar) = {
+  @transient private lazy val tuple = {
     val LambdaFunction(_, (elementVar: NamedLambdaVariable) +: tail, _) = function
     val indexVar = tail.headOption.map(_.asInstanceOf[NamedLambdaVariable])
     (elementVar, indexVar)
   }
+  @transient lazy val elementVar = tuple._1
```

Comment:
Hm, this pattern is a little ugly / adds extra overhead, but might be necessary vs just referencing "elementIndexVar._1" later in the code or something similar.